src/client/datascience/notebook/intellisense/pythonKernelCompletionProvider.ts

```diff
@@ -74,22 +75,15 @@ export class PythonKernelCompletionProvider implements CompletionItemProvider {
             traceError(`Live Notebook not available for ${getDisplayPath(notebookDocument.uri)}`);
             return [];
         }
-        const emptyResult: INotebookCompletion = { cursor: { end: 0, start: 0 }, matches: [], metadata: {} };
         // Allow slower timeouts for CI (testing).
         const timeout =
             parseInt(process.env.VSC_JUPYTER_IntellisenseTimeout || '0', 10) || Settings.IntellisenseTimeout;
         traceInfoIfCI(`Notebook completion request for ${document.getText()}, ${document.offsetAt(position)}`);
         const [result, pylanceResults] = await Promise.all([
-            Promise.race([
+            waitForPromise(
                 this.getJupyterCompletion(kernel.session, document.getText(), document.offsetAt(position), token),
-                sleep(timeout).then(() => {
-                    if (token.isCancellationRequested) {
-                        return;
-                    }
-                    traceInfoIfCI(`Notebook completions request timed out for Cell ${getDisplayPath(document.uri)}`);
```

Comment:
This was firing even after we already returned results. Tracing this messed me up. 

so using 'waitForPromise' which kills the timer.